.circleci/scripts.example/setup_hosts.sh

```diff
@@ -1,9 +1,11 @@
 #!/bin/bash
 set -e
 
+# Add host entries to match local docker development names.
 echo "Adding service hosts records"
 
 # These should match the services you are creating in your docker-compose file
+# This is required for containers to talk to each other
```

Comment:
```suggestion
# This is required for CircleCI to talk to containers
```

Reply:
can you explain this a bit more? in this instance does "CircleCI" mean a server (not a container) within circle's infrastructure that talks to our specified containers?

Reply:
@jpetto From my understanding, when a job is triggered on Circle CI, it provisions a VM (think your local machine) and runs the containers, exposing them on `localhost`. 

Before we run integration tests we [export the environment variables that our application depends on to the shell](https://github.com/Pocket/backend-typescript-template/blob/main/.circleci/config.yml#L159). This means that when tests are running and using configurations from our `config/index.ts` file (which pulls from these [env vars](https://github.com/Pocket/backend-typescript-template/blob/main/.docker/local.example.env#L12)), what it reads are the env var definitions for the docker container. CircleCI does not know what `mysql` means, so we tell CircleCI that when you see `mysql`, please look at it as `localhost` or `127.0.0.1`.

Now you may be asking, just as I'm asking myself now, that with projects like prospect-api that are set up to run outside the container without needing any of these [env vars](https://github.com/Pocket/backend-typescript-template/blob/main/.docker/local.example.env#L12), why do we export anything to the shell? I think the answer is that we don't need to and CIrcleCI should just work. I will confirm this after these short messages :)

Let me know if a live chat might help a bit more here.